src/other/Opportunity.js

```diff
@@ -0,0 +1,381 @@
+        SVGWidget.prototype.enter.apply(this, arguments);
+        var paddingTop = 30;
+        var nodeRectHeight = 14;
+        var verticalPadding = 10;
+        var h = (this.data().length + 1) * (nodeRectHeight + verticalPadding + 1) + paddingTop;
+        this.svg = element.append("g")
+            .attr("id", "opportunityID")
+            .attr("width", ((groupCount * 100) + 1))
+            .attr("height", h);
+    };
+    Opportunity.prototype.getIds = function() {
+        var dropdownList = this.columns();
+        dropdownList.unshift("default");
+        return dropdownList;
+    };
+    Opportunity.prototype.update = function(domNode, element) {
+        SVGWidget.prototype.update.apply(this, arguments);
+        var context = this;
+        console.log(this.data());
+        var data = this.data();
+        var dropDownOption = this.id();
+        var mouseHoverMapping = this.mouseHover();
+        data.sort(function(a, b) {
+            if (a.cur_group > b.cur_group) return 1;
+            else if (a.cur_group < b.cur_group) return -1;
+            else return 0;
+        });
+        var groups = [];
+        for (var i = 1; i <= groupCount; i++) {
+            groups.push(i);
+        }
+        var div = d3.select("body").append("div")
+            .attr("class", "tooltip")
+            .style("opacity", 0);
+        var div2 = d3.select("body").append("div2")
+            .attr("class", "tooltip")
+            .style("opacity", 0);
+        var paddingTop = 30;
+        var nodeRectHeight = 14;
+        var verticalPadding = 10;
+        var h = (data.length + 1) * (nodeRectHeight + verticalPadding + 1) + paddingTop;
+        var w = this.width(); //groupCount*100;
+        var nodeRectWidthPadding = 30;
+        var nodeRectWidth = ((w / groupCount) - nodeRectWidthPadding);
+        var group = this.svg.selectAll(".group").data(groups);
+        group.enter().append("rect")
+            .classed("group", true);
+        group.attr("width", function(d, i) {
+                switch (Math.abs(i + 1)) {
+                    case 0:
+                        return w / groupCount;
+                    case 1:
+                        return w / groupCount;
+                    default:
+                        return w / groupCount;
+                }
+            })
+            .attr("height", h - paddingTop)
+            .attr("x", function(d, i) {
+                switch (Math.abs(i + 1)) {
+                    case i:
+                        return (i * w / groupCount) + 1;
+                    default:
+                        return (i * w / groupCount) + 1;
+                }
+            })
+            .attr("y", paddingTop);
+        group.exit().remove();
+        d3.selectAll("#opportunityID g.update").remove();
+        var groupHeadings = this.svg.selectAll(".group_headings")
+            .data(groups);
+        groupHeadings.enter()
+            .append("text")
+            .classed("group_headings update", true)
```

Comment:
MInor:  ```.attr("class", "group_headings update")``` is the preferred approach when initializing the class (a little but faster).